### ISSUE TITLE
aws_ec2_managed_prefix_list should allow changes and resizing at the same time

### ISSUE THREAD

**Issue by jbscare** [hashicorp/terraform-provider-aws#22024]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Now that you can resize prefix lists, it'd be great if you could resize them and change their contents at the same time. This would require two API calls to AWS, because otherwise it says
```
InvalidParameterCombination: You cannot modify the entries and the maximum entries for the prefix list in the same request.
```
which is what it does now if you try.

### New or Affected Resource(s)

* aws_ec2_managed_prefix_list

### Potential Terraform Configuration

No configuration change; this is a request for an enhancement to an existing configuration.

### References

#20738

**Comment by kevindashton** [comment]
Agreed that order of operations is critical, but what if you're resizing down?  The API does not allow setting a max_entries value smaller than the current number of entries.  In that case, the actual entries would need to be updated _before_ the max_entries value.

This is a necessity for automated prefix lists where the max_entries value is based off a count() of a variable containing the entries.

**Comment by kevindashton** [comment]
likely resolved by #26845 and release [v4.32.0 of the Terraform AWS Provider](https://github.com/hashicorp/terraform-provider-aws/blob/v4.32.0/CHANGELOG.md).

**Comment by justinretzolk** [comment]
Closed by #26845
